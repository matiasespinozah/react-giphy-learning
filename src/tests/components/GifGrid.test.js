import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifts";
jest.mock('../../hooks/useFetchGifts')

describe('Prueba en <GifGrid />', () => {

  const category = 'dragon ball';
  

  test('Debe de mostrarse bien ', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
    const gifs = [{
      id: 'ABC',
      url: 'https://localhost//image.jpg',
      title: 'un titulo'
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
  
  
});
