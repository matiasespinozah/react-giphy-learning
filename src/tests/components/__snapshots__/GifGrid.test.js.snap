// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGrid /> Debe de mostrarse bien  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    dragon ball
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Loading... 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
